service/web/server_test.go

```diff
@@ -0,0 +1,113 @@
+package web
+
+import (
+	"context"
+	"testing"
+
+	metrov1 "github.com/razorpay/metro/rpc/proto/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetProjectIDFromRequest_PublishRequest(t *testing.T) {
```

Comment:
can refactor the tests as table-driven and re-use the assertions. Example below.
Refer https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

```
type test struct {
	input    interface{}
	expected string
	err      error
}

	tests := []test{
		{
			&metrov1.PublishRequest{
				Topic: "projects/project-p001/topics/topic-t001",
			}, "project-p001", nil,
		},
		{
			&metrov1.UpdateSubscriptionRequest{
				Subscription: &metrov1.Subscription{
					Name: "projects/project-p001/subscriptions/subscription-s001",
				},
			}, "project-p001", nil,
		},
	}

	for _, ts := range tests {
		pid, err := getProjectIDFromRequest(context.Background(), ts.input)
		assert.Equal(t, ts.expected, pid)
		assert.Equal(t, ts.err, err)
	}
```